### PR TITLE
Replace usages of auth config with app version

### PR DESCRIPTION
**Pull request by mpmadhavig**
### Related issues:
- https://github.com/wso2/product-is/issues/20880

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth.common/src/main/java/org/wso2/carbon/identity/oauth/common/OAuthConstants.java [hunk 1]
```diff
@@ -643,8 +643,6 @@ public static class OIDCConfigProperties {
         public static final String IS_SUBJECT_TOKEN_ENABLED = "isSubjectTokenEnabled";
         public static final String SUBJECT_TOKEN_EXPIRY_TIME = "subjectTokenExpiryTime";
         public static final int SUBJECT_TOKEN_EXPIRY_TIME_VALUE = 180;
-        public static final String IS_ACCESS_TOKEN_CLAIMS_SEPARATION_ENABLED =
```
Comment (DMHP) [inline components/org.wso2.carbon.identity.oauth.common/src/main/java/org/wso2/carbon/identity/oauth/common/OAuthConstants.java:646]: Is this added recently? Are we using a different constant for the server configuration?

File: components/org.wso2.carbon.identity.oauth.stub/src/main/resources/OAuthAdminService.wsdl [hunk 2]
```diff
@@ -395,7 +395,6 @@
                 <xs:sequence>
                     <xs:element minOccurs="0" name="OAuthVersion" nillable="true" type="xs:string"/>
                     <xs:element maxOccurs="unbounded" minOccurs="0" name="accessTokenClaims" nillable="true" type="xs:string"/>
-                    <xs:element minOccurs="0" name="accessTokenClaimsSeparationEnabled" type="xs:boolean"/>
```
Comment (DMHP) [inline components/org.wso2.carbon.identity.oauth.stub/src/main/resources/OAuthAdminService.wsdl:398]: Is this added very recently? We can remove such properties from the wsdl after adding

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/util/ClaimHandlerUtil.java [hunk 3]
```diff
@@ -44,4 +59,13 @@ private static boolean isAccessTokenClaimsSeparationFeatureEnabled() {
...
         return Boolean.parseBoolean(IdentityUtil.getProperty(ENABLE_CLAIMS_SEPARATION_FOR_ACCESS_TOKEN));
     }
+
+    private static boolean isAccessTokenClaimsSeparationEnabledForApp(OAuthAppDO oAuthAppDO, String tenantDomain)
+            throws IdentityOAuth2Exception {
+
+        ServiceProvider serviceProvider = OAuth2Util.getServiceProvider(oAuthAppDO.getOauthConsumerKey(), tenantDomain);
+        String appVersion = serviceProvider.getApplicationVersion();
+
+        return OAuth2Util.isGivenAppVersionAllowed(appVersion, ApplicationConstants.ApplicationVersion.APP_VERSION_V2);
```
Comment (DMHP) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/util/ClaimHandlerUtil.java:69]: isGivenAppVersionAllowed > isAppVersionAllowed

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/util/ClaimHandlerUtil.java [hunk 4]
```diff
@@ -30,11 +35,21 @@
...
+    public static CustomClaimsCallbackHandler getClaimsCallbackHandler(OAuthAppDO oAuthAppDO)
+            throws IdentityOAuth2Exception {
 
         // If JWT access token OIDC claims separation is enabled and the application is configured to separate OIDC
         // claims, use the JWTAccessTokenOIDCClaimsHandler to handle custom claims.
-        if (isAccessTokenClaimsSeparationFeatureEnabled() && oAuthAppDO.isAccessTokenClaimsSeparationEnabled()) {
+        int appTenantId = IdentityTenantUtil.getLoginTenantId();
+        String tenantDomain = IdentityTenantUtil.getTenantDomain(appTenantId);
+        if (isAccessTokenClaimsSeparationFeatureEnabled() &&
+                isAccessTokenClaimsSeparationEnabledForApp(oAuthAppDO, tenantDomain)) {
```
Comment (DMHP) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/util/ClaimHandlerUtil.java:52]: Can   isAccessTokenClaimsSeparationEnabledForApp become true even if isAccessTokenClaimsSeparationFeatureEnabled() false. If so can we please block that?

File: components/org.wso2.carbon.identity.oauth.stub/src/main/resources/OAuthAdminService.wsdl (hunk 2 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.oauth.stub/src/main/resources/OAuthAdminService.wsdl:398]: Introduced with @shashimalcse feature. Removing this as we do not want to keep this variable in the access token data structure anymore.

File: components/org.wso2.carbon.identity.oauth.common/src/main/java/org/wso2/carbon/identity/oauth/common/OAuthConstants.java (hunk 1 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.oauth.common/src/main/java/org/wso2/carbon/identity/oauth/common/OAuthConstants.java:646]: This is the sp oidc property name.

`OAuth.OpenIDConnect.EnableClaimsSeparationForAccessToken` is the server level config name.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/util/ClaimHandlerUtil.java (hunk 3 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/util/ClaimHandlerUtil.java]: 49ed7df33aadc715204bb04ff558177e3a777a72

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/util/ClaimHandlerUtil.java (hunk 4 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/util/ClaimHandlerUtil.java:52]: These two are server and app level enabling configs.

Server level -> `isAccessTokenClaimsSeparationFeatureEnabled`
App level -> ` isAccessTokenClaimsSeparationEnabledForApp`

**Comment by mpmadhavig** [comment]
Integration failures will be fixed with https://github.com/wso2/product-is/pull/21446

**Review by DMHP — APPROVED** [review]

**Pull request merged**